### ISSUE TITLE
GetSymbolInfo() synthesizes built-in operator methods on System.IntPtr rather than nint

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#42524]
`SemanticModel.GetSymbolInfo()` returns a synthesized symbol with containing type `System.IntPtr` rather than `nint` for built-in unary operators on native integers.

For instance, with `+operand` below:
```C#
class Program
{
    static System.IntPtr Evaluate(System.IntPtr operand)
    {
        return +operand;
    }
}
```
Expected: `nint nint.op_UnaryPlus(nint value)`
Actual: `nint System.IntPtr.op_UnaryPlus(System.IntPtr value)`

**Comment by AlekseyTs** [comment]
@cston It doesn't look like the repro scenario uses nint. Why would we expect nint to show up in the picture at all?

**Comment by cston** [comment]
> Why would we expect nint to show up in the picture at all?

It depends if operations on `IntPtr` use the built-in `nint` operations. If so, then the expected result is `nint System.IntPtr.op_UnaryPlus(System.IntPtr value)`. Otherwise the compiler should report `error CS0023: Operator '+' cannot be applied to operand of type 'IntPtr'`.

**Comment by cston** [comment]
Compiler now reports:
```
(5,16): error CS0023: Operator '+' cannot be applied to operand of type 'IntPtr'
```
